### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by initialneil**
- add `opencv_imgcodecs` to `target_link_libraries` for OpenCV 3.0 
  - fixing error:
```
    /usr/bin/ld: libBING.a(Objectness.cpp.o): undefined reference to symbol     '_ZN2cv7imwriteERKNS_6StringERKNS_11_InputArrayERKSt6vectorIiSaIiEE'
    /usr/local/lib/libopencv_imgcodecs.so.3.0: error adding symbols: DSO missing from command line
```

- re-organize library compile order 
  - previously the `LibLinear` was compiled as library but not used; the `BING` was not considered as library
  - now `LibLinear` and `BING` are compiled as libraries and linked to a single `main.cpp`

### CODE REVIEW

**Comment by RyanCV** [comment]
this works, thanks!

**Comment by RyanCV** [comment]
@initialneil , after compile, I have the following problems when run sudo./BING_linux, do you know how to solve it?? Thanks.
Invalidate class name
 in /media/BING_Linux/Src/DataSetVOC.cpp:125
OpenCV Error: Assertion failed (Invalidate class name
) in loadBox, file /media/BING_Linux/Src/DataSetVOC.cpp, line 125
terminate called after throwing an instance of 'cv::Exception'
  what():  /media/BING_Linux/Src/DataSetVOC.cpp:125: error: (-215) Invalidate class name
 in function loadBox

**Comment by initialneil** [comment]
Hi, @RyanCV . [DataSetVOC.cpp line 125](https://github.com/initialneil/Objectness/blob/master/Src/DataSetVOC.cpp#L125):
```
    CV_Assert_(clsIdx[clsIdx.size() - 1] >= 0, ("Invalidate class name\n"));
```
means either `clsIdx.size() == 0`(there was no class loaded from the ground truth) or `clsIdx[...] < 0`(class labels were not valid). I guess your training data was not well prepared.

**Comment by RyanCV** [comment]
@initialneil, actually the error is caused by the different txt format in windows and Linux. I create a new class.txt file in linux and the problem has been solved. Thanks for your information.

**Comment by bittnt** [comment]
Thanks a lot for updating this with OpenCV3.0!
